src/Sarif.Multitool/Rules/MessagesShouldEndWithPeriod.cs

```diff
@@ -29,68 +34,56 @@ protected override IEnumerable<string> FormatIds
+        protected override void Analyze(Rule rule, string rulePointer)
         {
-            Analyze(result.Message, resultPointer);
+            AnalyzeMessageStrings(rule.MessageStrings, rulePointer, SarifPropertyName.MessageStrings);
+            AnalyzeMessageStrings(rule.RichMessageStrings, rulePointer, SarifPropertyName.RichMessageStrings);
         }
 
-        protected override void Analyze(Rule rule, string rulePointer)
+        private void AnalyzeMessageStrings(
+            IDictionary<string, string> messageStrings,
+            string rulePointer,
+            string propertyName)
         {
-            if (rule.MessageFormats != null)
+            if (messageStrings != null)
             {
-                foreach (string formatId in rule.MessageFormats.Keys)
+                foreach (string key in messageStrings.Keys)
                 {
-                    string messageFormat = rule.MessageFormats[formatId];
-                    if (DoesNotEndWithPeriod(messageFormat))
+                    string messageString = messageStrings[key];
+                    if (!String.IsNullOrEmpty(messageString) && DoesNotEndWithPeriod(messageString))
                     {
                         string messagePointer = rulePointer
-                            .AtProperty(SarifPropertyName.MessageFormats)
-                            .AtProperty(formatId);
+                            .AtProperty(propertyName)
+                            .AtProperty(key);
 
                         LogResult(
                             messagePointer,
                             nameof(RuleResources.SARIF1008_Default),
-                            messageFormat);
+                            messageString);
                     }
                 }
             }
         }
 
-        protected override void Analyze(Stack stack, string stackPointer)
-        {
-            Analyze(stack.Message, stackPointer);
-        }
-
-        protected override void Analyze(StackFrame frame, string framePointer)
+        protected override void Analyze(Message message, string messagePointer)
         {
-            Analyze(frame.Message, framePointer);
+            AnalyzeMessageString(message.Text, messagePointer, SarifPropertyName.Text);
+            AnalyzeMessageString(message.RichText, messagePointer, SarifPropertyName.RichText);
         }
 
-        private void Analyze(string message, string pointer)
+        private void AnalyzeMessageString(
+            string messageString,
+            string messagePointer,
+            string propertyName)
         {
-            if (DoesNotEndWithPeriod(message))
+            if (!String.IsNullOrEmpty(messageString) && DoesNotEndWithPeriod(messageString))
             {
-                string messagePointer = pointer.AtProperty(SarifPropertyName.Message);
+                string textPointer = messagePointer.AtProperty(SarifPropertyName.Text);
```

Comment:
>SarifPropertyName.Text [](start = 63, length = 22)

BUG: This should be `propertyName` (I missed this in the refactor), so I'm missing a unit test case.

Reply:
Fixed in Iteration 8.

---
In reply to: [209376585](https://github.com/Microsoft/sarif-sdk/pull/1009#discussion_r209376585) [](ancestors = 209376585)